testplan/testing/base.py

```diff
@@ -628,74 +657,84 @@ def process_test_data(self, test_data):
         """
         raise NotImplementedError
 
-    def get_process_failure_report(self):
+    def get_process_check_report(self, retcode, stdout, stderr):
         """
         When running a process fails (e.g. binary crash, timeout etc)
         we can still generate dummy testsuite / testcase reports with
         a certain hierarchy compatible with exporters and XUnit conventions.
         """
-        from testplan.testing.multitest.entries.assertions import RawAssertion
-
-        assertion_content = os.linesep.join(
+        assertion_content = "\n".join(
             [
-                "Process failure: {}".format(self.cfg.driver),
-                "Exit code: {}".format(self._test_process_retcode),
-                "stdout: {}".format(self.stdout),
-                "stderr: {}".format(self.stderr),
+                "Process: {}".format(self.cfg.driver),
+                "Exit code: {}".format(retcode),
             ]
         )
 
+        passed = retcode == 0 or retcode in self.cfg.ignore_exit_codes
+
         testcase_report = TestCaseReport(
-            name="failure",
+            name="ExitCodeCheck",
+            uid="ExitCodeCheck",
+            suite_related=True,
             entries=[
                 RawAssertion(
-                    description="Process failure details",
+                    description="Process exit code check",
                     content=assertion_content,
-                    passed=False,
-                ).serialize()
+                    passed=passed,
+                ).serialize(),
+                Log(
+                    message=stdout.read(), description="Process stdout",
+                ).serialize(),
+                Log(
+                    message=stderr.read(), description="Process stderr",
+                ).serialize(),
             ],
         )
 
-        testcase_report.status_override = Status.ERROR
+        testcase_report.runtime_status = RuntimeStatus.FINISHED
 
-        return TestGroupReport(
-            name="ProcessFailure",
-            category="testsuite",
+        suite_report = TestGroupReport(
+            name="ProcessChecks",
+            category=ReportCategories.TESTSUITE,
             entries=[testcase_report],
         )
 
+        return suite_report
+
     def update_test_report(self):
         """
         Update current instance's test report with generated sub reports from
         raw test data. Skip report updates if the process was killed.
         """
-        if self._test_process_killed or not self._test_has_run:
-            self.result.report.append(self.get_process_failure_report())
-            return
-
-        if len(self.result.report):
-            raise ValueError(
-                "Cannot update test report,"
-                " it already has children: {}".format(self.result.report)
-            )
+        with open(self.stdout) as stdout, open(self.stderr) as stderr:
```

Comment:
Could `stdout` / `stderr` not be a filename? Like, could it ever be a `io.StringIO`?

Reply:
No I don't think so. See their definition: https://github.com/Morgan-Stanley/testplan/blob/b2725e2f2761e4991558faa50d375032cbf03ba0/testplan/testing/base.py#L444-L450